src/App.tsx

```diff
@@ -17,6 +50,10 @@ export const App: React.FC = () => {
             <div className="control">
               <input
                 type="text"
+                defaultValue={query}
+                onChange={(event) => {
+                  handleSet(event.target.value);
```

Comment:
could be easier to change the value from the setQuery function, the setQuery is already a function that changes the value..

Reply:
totally agree with @alaaashkar - if handler just invokes the setter - just use the setter straight away
```suggestion
                  setQuery(event.target.value);
```